### PR TITLE
Fix printing wrapper geoms

### PR DESCRIPTION
**Pull request by BenCurran98**
Proposing a fix for [this issue](https://github.com/JuliaGeo/GeoInterface.jl/issues/126) where we modify the displayed text for our wrapper geometries. Each geometry can be displayed in "regular" or "compact" form which will influence how much information is shown to the user. This should make things a bit clearer so we don't get spammed with the raw fields that can be hard to follow.

### CODE REVIEW

**Comment by asinghvi17** [comment]
This looks great, thanks!  Love the compactification of the elements you have going on there. 

One more thing that I'm thinking we should do is to decrease the number of points printed, so that that any geometry fits on one line.  Otherwise you get crazy long strings, aka what we have now, which have crashed my terminal a couple times when ultra long geometries are considered (10k points or higher).

Consider something like the following:

```julia
screen_nrows, screen_ncols = displaysize(io)
# limit the number of points plotted to 
# floor((screen_ncols - currently_used_space - 1 #= triple dot character =#) / length_of_one_point_in_chars)
# add a `...` in between (unicode character) or indicate number of points somehow, as is done in Strings
```

I will post a more useful prototype sometime later today, but this is the basic concept.

**Comment by BenCurran98** [comment]
@asinghvi17 I've had a crack at this, is this sort of what you're after? Now we compress geometries with "..." and report the number of sub geometries hidden, e.g.
```julia
linestring = GI.LineString(map(_ -> rand(2), 1:10_000))
```
Gives
```julia
LineString{false, false}([[0.6022716265309537, 0.46854202492808894], [0.7698994281922478, 0.5141934816807233],  … (9997) … [0.5275211157024845, 0.7022242690985355], [0.09265052455532974, 0.5388384079940159]])
```

**Comment by asinghvi17** [comment]
Sorry for the delay!  This is next on my list.  Should get to it by Wednesday.

**Comment by rafaqz** [comment]
Where is this one at? would be nice to merge

**Comment by asinghvi17** [comment]
I'm using it locally but have a few fixes I need to push.  Other than that it's great!

**Comment by BenCurran98** [comment]
Hey @asinghvi17 , sorry for the delay (been super swamped at work lately)
I've updated some of the logic here to be a bit nicer with how we cut down the displayed details with nested geom types (like polygons with linear rings, etc.) so now we cut a bit out of each sub geometry recursively, e.g.

```julia
linearring = GI.LinearRing([(1, 2), (3, 4), (5, 6), (1, 2)])
polygon = GI.Polygon([linearring, linearring])

# gives "Polygon{false, false}([LinearRing([(1, 2), … (2) … , (1, 2)]), LinearRing([(1, 2), … (2) … , (1, 2)])])"
```

File: src/wrappers.jl [hunk 1]
```diff
@@ -228,6 +301,45 @@ for (geomtype, trait, childtype, child_trait, length_check, nesting) in (
     end
 end
 
+function _nice_geom_str(geom, ::Bool, ::Bool, ::Int) 
+    io = IOBuffer()
+    show(io, MIME("text/plain"), geom)
+    return String(take!(io))
+end
+
```
Comment (rafaqz) [inline src/wrappers.jl:309]:
```suggestion
```

File: src/wrappers.jl [hunk 2]
```diff
@@ -228,6 +301,45 @@ for (geomtype, trait, childtype, child_trait, length_check, nesting) in (
...
+# handle tuples/vectors explicitly
+function _nice_geom_str(geom::AbstractVector, ::Bool, compact::Bool, ::Int)
+    spacing = compact ? "" : " "
+    str = "["
+    str *= _add_elements_with_spacing(geom, spacing)
+    str *= "]"
+    return str
+end
+
```
Comment (rafaqz) [inline src/wrappers.jl:326]:
```suggestion
```

File: src/wrappers.jl [hunk 3]
```diff
@@ -228,6 +301,45 @@ for (geomtype, trait, childtype, child_trait, length_check, nesting) in (
...
+# need a work around to pass the show_mz variable through - put string to a temp IOBuffer then read it
+function _nice_geom_str(geom::WrapperGeometry, show_mz::Bool, compact::Bool, screen_ncols::Int) 
+    buf = IOBuffer()
+    io = IOContext(IOContext(buf, :compact => compact))
+    show(io, MIME("text/plain"), geom; show_mz = show_mz, screen_ncols = screen_ncols)
+    return String(take!(buf))
+end
+
```
Comment (rafaqz) [inline src/wrappers.jl:317]:
```suggestion
```

File: src/wrappers.jl [hunk 4]
```diff
@@ -377,6 +519,36 @@ function Feature(geometry=nothing; properties=nothing, crs=nothing, extent=nothi
     end
 end
 
+function Base.show(io::IO, ::MIME"text/plain", f::Feature; show_mz::Bool = true)
+    compact = get(io, :compact, false)
+    spacing = compact ? "" : " "
+    print(io, "Feature(")
+    Base.show(io, MIME("text/plain"), f.parent.geometry; show_mz = show_mz)
```
Comment (rafaqz) [inline src/wrappers.jl:526]:
```suggestion
    show(io, MIME("text/plain"), f.parent.geometry; show_mz = show_mz)
```

File: src/wrappers.jl [hunk 5]
```diff
@@ -377,6 +519,36 @@ function Feature(geometry=nothing; properties=nothing, crs=nothing, extent=nothi
...
+    non_geom_props = filter(!=(:geometry), propertynames(f.parent))
+    if !isempty(non_geom_props)
+        print(io, ",$(spacing)properties$(spacing)=$(spacing)(")
+        for (i, property) ∈ enumerate(non_geom_props)
+            print(io, "$(property)$(spacing)=$(spacing)")
+            Base.show(io, getproperty(f.parent, property))
```
Comment (rafaqz) [inline src/wrappers.jl:532]:
```suggestion
            show(io, getproperty(f.parent, property))
```

**Review by rafaqz — APPROVED** [review]
Looks good to me, lets merge. 

Comments are just simple code cleanup
